src/wasm/kotlin/java/arcs/Addressable.kt

```diff
@@ -0,0 +1,30 @@
+package arcs
+
+/**
+ * Any object implementing this interface can be converted into a stable identifier. This
+ * could a platform specific pointer, identity hashcode, or id as long as it is guaranteed unique.
+ */
+interface Addressable
+
+typealias Address = Int
+
+internal val address2Addressable = kotlin.collections.mutableMapOf<Address, Addressable>()
+internal val addressable2Address = kotlin.collections.mutableMapOf<Addressable, Address>()
+
+internal var nextAddress = 1;
+
+fun Addressable.toAddress(): Address {
```

Comment:
Technically it should be impossible for an address other than '0' to be a null, some wasm.ts APIs send '0' as a null. Do we need it? I'm not sure, this may only be for WasmStrings, perhaps Particles and Handles can never be null, and therefore value to find the address in the map lookup should be an exception?